src/Assets.php

```diff
@@ -187,8 +188,14 @@ public static function get_wc_block_data( $settings ) {
 					'privacy'  => self::format_page_resource( $page_ids['privacy'] ),
 					'terms'    => self::format_page_resource( $page_ids['terms'] ),
 				],
-				'checkoutAllowsGuest'           => filter_var( get_option( 'woocommerce_enable_guest_checkout' ), FILTER_VALIDATE_BOOLEAN ),
-				'checkoutAllowsSignup'          => filter_var( get_option( 'woocommerce_enable_signup_and_login_from_checkout' ), FILTER_VALIDATE_BOOLEAN ),
+				'checkoutAllowsGuest'           => $checkout instanceof \WC_Checkout && false === filter_var(
+					$checkout->is_registration_required(),
+					FILTER_VALIDATE_BOOLEAN
+				),
+				'checkoutAllowsSignup'          => $checkout instanceof \WC_Checkout && filter_var(
+					$checkout->is_registration_enabled(),
+					FILTER_VALIDATE_BOOLEAN
+				),
```

Comment:
Could consider renaming the block option const to more closely match the API. Probably no need to do this now, not a blocker.